Sources/VSM/Documentation.docc/ComprehensiveGuide/ModelDefinition.md

```diff
@@ -184,134 +183,134 @@ First, we'll take a look at the shape of the `EditUserProfileViewState` and its
+After we have a good idea of how the states and data should flow, we can begin implementing the editing model of the `EditUserProfileViewState` by creating the corresponding model struct.
 
 ```swift
-extension EditUserProfileViewState.EditingModel {
+struct EditingModel: EditingModeling {
+    let userData: UserData
+
     init(userData: UserData) {
-        saveUsername = { username in
-            return Just(
-                EditUserProfileViewState(
-                    data: userData,
-                    editingState: .saving
-                )
+        self.userData = userData
+    }
+
+    func save(username: String) -> AnyPublisher<EditUserProfileViewState, Never> {
+        return Just(
+            EditUserProfileViewState(
+                data: userData,
+                editingState: .saving
             )
-            .merge(with: Self.save(username: username, for: userData))
-            .eraseToAnyPublisher()
-        }
+        )
+        .merge(with: performSave(username: username))
+        .eraseToAnyPublisher()
     }
 }
 ```
 
-Our repository needs a `UserData` object because the save function also handles the `retry()` and `cancel()` actions in the event of an error. So, the initializer will ask for the user data so that we can forward it to the save function within the `saveUsername()` closure.
+Our repository needs a `UserData` object because the save function also handles the `retry()` and `cancel()` actions in the event of an error. So, the initializer will ask for the user data so that we can forward it to the `performSave()` function.
 
-Similar to the load action from the Load Profile model, we'll immediately return a new `.saving` state to the view while the save operation is processing. Notice how we have to recreate the view state struct to do so.
+Similar to the load action from the Load Profile model, we'll immediately return a new "saving" state to the view while the save operation is processing. Notice how we have to recreate the view state struct to do so.
 
-Next, we'll implement the save function like so:
+Next, we'll implement the `performSave()` function like so:
 
 ```swift
-private static func save(
-    username: String,
-    for userData: UserData
-) -> AnyPublisher<EditUserProfileViewState, Never> {
-    UserDataRepository().saveUsername(username)
-        .map { savedUserData in
-            EditUserProfileViewState(
-                data: savedUserData,
-                editingState: .editing(EditUserProfileViewState.EditingModel(userData: savedUserData))
-            )
-        }
-        .catch { error in
-            handleSavingError(error, username: username, for: userData)
-        }
-        .eraseToAnyPublisher()
+struct EditingModel: EditingModeling {
+    ...
+    private func performSave(username: String) -> AnyPublisher<EditUserProfileViewState, Never> {
+        UserDataRepository().save(username: username)
```

Comment:
Is your use of `UserDataRecpository()...` just a convenience for this example? Shouldn't that be an injected object or something? 😄 

I guess my concern is that someone reads this and decides this sort of inline construction of something like a data repository is a good thing to do... (or at least "acceptable" 😨 )

Reply:
It is done this way to simplify the example. To reinforce dependency injection, the first occurrence of this phenomenon on lines 74-92 was specifically called out with:

> (We will cover proper dependency injection in <doc:DataDefinition>.)

Do you feel like this needs another direct callout here as well?

Reply:
There are technically 4 locations, but those are grouped into two examples which are reiterated across two code samples. I'll add a small snippet next to the second example.

Reply:
👍

Reply:
Addressed. Please take a look at line 232

Reply:
Awesome. Thanks!